include/lo2s/shared_memory.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of the lo2s software.
+ * Linux OTF2 sampling
+ *
+ * Copyright (c) 2016,
+ *    Technische Universitaet Dresden, Germany
+ *
+ * lo2s is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * lo2s is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with lo2s.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <lo2s/error.hpp>
+
+#include <utility>
+
+extern "C"
+{
+#include <fcntl.h>
+#include <sys/mman.h>
+}
+
+namespace lo2s
+{
+class SharedMemory
+{
+public:
+    SharedMemory() : addr_(nullptr), size_(0)
+    {
+    }
+
+    SharedMemory(SharedMemory&) = delete;
+    SharedMemory& operator=(SharedMemory&) = delete;
+
+    SharedMemory(SharedMemory&& other)
+    {
+        addr_ = other.addr_;
+        size_ = other.size_;
+        other.addr_ = nullptr;
+    }
+
+    SharedMemory& operator=(SharedMemory&& other)
+    {
+        addr_ = other.addr_;
+        size_ = other.size_;
+        other.addr_ = nullptr;
```

Comment:
This is not correct behavior. Move-Assignment has to work for two "valid" objects. You can't just simply discard one mapping.

Using std::swap was one correct implementation. Another correct solution would be to `munmap(addr_)`  if it is not the `nullptr` before overwriting it with `other.addr_`.